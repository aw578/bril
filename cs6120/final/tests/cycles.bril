@main {
  a: int = const 0;
  one: int = const 1;
  ten: int = const 10;
  x: int = id a;
  y: int = const 0;
  not_y: int = add x one;
  not_c: bool = lt y ten;

.loop:
  y: int = add x one;
  c: bool = lt y ten;
  br c .then .end;

.then:
  x: int = id y;
  jmp .loop;

.end:
  ret;
}