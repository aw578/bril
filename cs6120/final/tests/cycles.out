Congruence classes for function main:
Class const 0: ['a.0', 'y.0']
Class const 1: ['one.0']
Class const 10: ['ten.0']
Class id: ['x.0']
Class add: ['not_y.0']
Class lt: ['not_c.0']
Class phi: ['x.1']
Class add_0: ['y.2']
Class id_0: ['x.2']
Class lt_0: ['c.1']
Class phi_0: ['y.1']
Class phi_1: ['c.0']
